src/App/Fig.purs

```diff
@@ -162,27 +209,27 @@ loadFig spec@{ fluidSrcPaths, inputs, imports, file, datasets } = do
       gcFwd :: Env 𝔹 -> Val 𝔹 × GraphImpl
       gcFwd γ = graphgc_op.bwd (deMorgan focusFwd γ)
 
-      in_views = mapWithKey (\_ _ -> Nothing) (unwrap γ)
+      in_views = const Nothing <$> (unwrap γ_restricted)
 
       γ0 = botOf γα :: Env 𝔹
       v0 = botOf outα :: Val 𝔹
 
       inertBwd = vertices g0 \\ (vertices $ snd (gcBwd (topOf outα))) :: Set DVertex
-      inertFwd = (vertices $ snd $ (graphgc.fwd <<< focusFwd) γ0)
+      inertFwd = vertices $ snd $ (graphgc.fwd <<< focusFwd) γ0
 
       γInert = selState <$> select𝔹s γα inertBwd :: Env (𝔹 -> 𝔹 -> SelState 𝔹)
       vInert = selState <$> select𝔹s outα inertFwd :: Val (𝔹 -> 𝔹 -> SelState 𝔹)
 
-      v' = lift γInert gcBwd
-      γ' = lift vInert gcFwd
+      vf = lift γInert gcBwd
+      γf = lift vInert gcFwd
 
       linkedInputs :: Env (SelState 𝔹) -> Env (SelState 𝔹) × Val (SelState 𝔹) × Selection GraphImpl × Set DVertex
-      linkedInputs = γ' >>> \(v × g × g') -> (fst $ v' v) × v × { persistent: g, transient: g' } × inertFwd
+      linkedInputs = γf >>> \(v × g × g') -> (fst $ vf v) × v × { persistent: g, transient: g' } × inertFwd
 
       linkedOutputs :: Val (SelState 𝔹) -> Val (SelState 𝔹) × Env (SelState 𝔹) × Selection GraphImpl × Set DVertex
-      linkedOutputs = v' >>> \(γ × g × g') -> (fst $ γ' γ) × γ × { persistent: g, transient: g' } × inertBwd
+      linkedOutputs = vf >>> \(γ × g × g') -> (fst $ γf γ) × γ × { persistent: g, transient: g' } × inertBwd
 
-   pure { spec, s, γ: γInert <*> γ0 <*> γ0, v: vInert <*> v0 <*> v0, linkedOutputs, linkedInputs, dir: LinkedOutputs, in_views, out_view: Nothing }
+   pure { spec, s, γ: γInert <*> γ0 <*> γ0, v: vInert <*> v0 <*> v0, linkedOutputs, linkedInputs, dir: LinkedOutputs, in_views, out_view: Nothing, intermediate_views: Dict empty, intermediate_values: Dict empty, in_roots }
```

Comment:
Could be time to split this over multiple lines -- this function is pretty heavyweight anyway.